filebeat/harvester/reader/json_test.go

```diff
@@ -304,6 +305,13 @@ func TestAddJSONFields(t *testing.T) {
 			},
 			ExpectedTimestamp: time.Time{},
 		},
+		{
+			// if document_id is set, extract the ID from the event
+			Name:       "extract event id",
+			Data:       common.MapStr{"@timestamp": common.Time(now), "json": common.MapStr{"id": "test_id"}},
```

Comment:
What happen if `id` is not a string?

Reply:
Adde a test for "id" not being a string.
Reading ID is similar to reading "MessageKey". If it's missing we return an empty string.